gulp-tasks/stageGitHub/determineDeploymentType.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Gets a list of changed files by looking at the currently
+ * checked out commit in Google Cloud Build and either outputs 'static'
+ * or 'app', depending on the defined globs which are matched against
+ * the list of changed files.
+ */
+
+const micromatch = require('micromatch');
+const fs = require('fs/promises');
+const path = require('path');
+
+const {fetchGitHubApi} = require('./lib/fetchGitHubApi');
+
+const APP_GLOB = ['package.json', 'server/**/*.js'];
+const STATIC_GLOB = ['site/**/*'];
+
+const OUTPUT_APP_BUILD = 'app';
+const OUTPUT_STATIC_BUILD = 'static';
+
+const DEPLOYMENT_TYPE_PATH = path.join(__dirname, 'tmp', 'deploymentType.txt');
+
+async function determineDeploymentType() {
+  const prNumber = process.env.PR_NUMBER;
+  if (!prNumber) {
+    console.warn(
+      'This task is inteded to run on Google Cloud Build, which exports $PR_NUMBER. ' +
+        'Use npm run stage:personal locally instead.'
+    );
+    return;
+  }
+
+  let changedFiles = await fetchGitHubApi(
```

Comment:
Since this runs on cloud, does it know github_token variable?

Reply:
ah, i see it is passed in the workflow file now

Reply:
Yes, I added a token to secret manager. See changes in .cloudbuild/stage.yaml.